### ISSUE TITLE
Odd behavior with importing typevars from other files

### ISSUE THREAD

**Issue by danielgordon10** [microsoft/pyright#1873]
Pyright version 1.1.140
file1.py source
```python
from typing import TypeVar
class A:
    pass
class B:
    pass

T = TypeVar("T", A, B)
```
file2.py source
```python
from file1 import A, T

def foo(val: T) -> T:
    return val

def bar(val: A) -> A:
    out = foo(val)
    return out  # results in  Expression of type "T" cannot be assigned to return type "A"`

```

But if T is defined in file2.py everything works as expected.

**Comment by erictraut** [comment]
Hmm, I'm not able to repro the problem with these samples. Are you sure you're able to repro with this minimal example, or is there perhaps something more involved?

Is it possible that you have another `file1` (e.g. a `file1.pyi` stub file) present? If you select `T` or `A` in `file2.py` and choose "Go To Declaration" (right click in VS Code), does it take you to `file1.py`?

**Comment by danielgordon10** [comment]
Try this one
file2.py
```python
import file1
def foo(val: file1.T) -> file1.T:
    return val

def bar(val: file1.A) -> file1.A:
    out = foo(val)
    return out  # results in  Expression of type "T" cannot be assigned to return type "A"  Type "A | B" cannot be assigned to type "A"    "B" is incompatible with "A"
```

**Comment by erictraut** [comment]
Thanks, that I was able to repro the problem and identify the cause. The code wasn't properly dealing with TypeVars that were imported from another file and referenced using member access expressions (e.g. `file1.A`). 

The fix will be in the next release. Thanks again for the bug report.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.141, which I just published. It will also be included in the next release of pylance.
